### PR TITLE
Front end Development Updates

### PR DESCRIPTION
**Pull request by Ciezo**
<h2>Updates to the View Layer</h2>

- Controller now linked to View
- Various methods of controller is now accessible to the View layer which should be helpful in debugging
- Music records now retrievable and updates with the app
- New music records to be inserted to the MySQL server now also dynamically updates locally using:<br> `private static void music_lib_refresher(String music_title, String music_artist, String music_album)`
- Viewing lyrics according to what is played 
- Fix some bugs

### CODE REVIEW

File: src/com/jester/controller/Controller.java [hunk 1]
```diff
@@ -50,7 +50,7 @@ public Music SetNewMusic(Music insert_music) {
     }
     
     public String[] GetAllStoredAlbums() {
-        String[] albums = new String[5];
+        String[] albums = new String[1000];
```
Comment (cloydsecuya) [inline src/com/jester/controller/Controller.java:53]: Updated the size to prevent issues with scaling

File: src/com/jester/view/View.java [hunk 2]
```diff
@@ -25,6 +40,12 @@
 public class View {
     private static final Border EmptyBorder = null;
     
+    static Controller controller = new Controller();
+    static Music[] music_ls; 
+    static Music music; 
+    static DefaultTableModel model = new DefaultTableModel();
```
Comment (cloydsecuya) [inline src/com/jester/view/View.java:46]: Expose the table model to make it accessible with the music library refresher

File: src/com/jester/view/View.java [hunk 3]
```diff
@@ -169,18 +203,44 @@ public void valueChanged(ListSelectionEvent event) {
...
+                
+                // Set up and assign the rest of the attributes
+                music.setMusic_title(new_music_title);
+                music.setMusic_artist(new_music_artist);
+                music.setMusic_album(new_music_album);
+                music.setMusic_path_to_DIR(new_music_path);
+                // Final instantiate to attributes
+                music = new Music(music.getMusic_ID(), new_music_title, new_music_artist, new_music_album, new_music_path);
+                // Call the controller to insert the records
+                controller.SetNewMusic(music);
```
Comment (cloydsecuya) [inline src/com/jester/view/View.java:241]: Inserts the new music records into the database. 
Includes all Music class members

File: src/com/jester/view/View.java [hunk 4]
```diff
@@ -169,18 +203,44 @@ public void valueChanged(ListSelectionEvent event) {
...
+                // Refresh the library for new contents
+                music_lib_refresher(music.getMusic_title(), music.getMusic_artist(), music.getMusic_album());
```
Comment (cloydsecuya) [inline src/com/jester/view/View.java:243]: Refresh the table model locally for updating the records in the App locally

File: src/com/jester/view/View.java [hunk 5]
```diff
@@ -325,20 +385,33 @@ public void actionPerformed(ActionEvent e) {
         ImageIcon scaledSB = new ImageIcon(SBscale);
         ImageIcon scaledPB = new ImageIcon(PBscale);
         playButtonContainer.setIcon(scaledPB);
+        
+        // TODO: Play and Stop music button
         playButtonContainer.addMouseListener(new MouseAdapter() {
             private boolean playing = false;
```
Comment (cloydsecuya) [inline src/com/jester/view/View.java:391]: This one is causing some problems due to it being only assigned upon Mouse event notification

**Review by cloydsecuya — APPROVED** [review]

File: src/com/jester/controller/Controller.java (hunk 1 above)
Comment (Ciezo) [reply src/com/jester/controller/Controller.java:53]: This one is for making the return more viable.

**Pull request merged**